src/server/req_message.c

```diff
@@ -204,43 +204,14 @@ req_py_spawn(struct batch_request *preq)
 		return;
 	}
 
-	if (jt == IS_ARRAY_NO) {		/* a regular job is okay */
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {		/* a regular job or a subjob is okay */
 		/* the job must be running */
 		if ((pjob->ji_qs.ji_state != JOB_STATE_RUNNING) ||
 			(pjob->ji_qs.ji_substate !=
 			JOB_SUBSTATE_RUNNING)) {
```

Comment:
Earlier to this code change if subjob is in state JOB_STATE_RUNNING and its substate is in JOB_SUBSTATE_PRERUN this combination was allowed.  Any reason why this is not allowed now?

Reply:
Now subjob and normal job will be treated same here.  If you have any use case in mind for not allowing this combination for subjob please point me to it.